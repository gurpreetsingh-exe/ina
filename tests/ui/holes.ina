// COMPILE_FAIL

extern mod core;
using core::prelude::*;

fn id[T](v: T) -> T { v }

fn main() {
    let xs = [None, Some(_)];
    let _: [Option[i64]] = id(xs);
    let x = xs[0];
    let _ = [x, _];
    match Some![i8](20) {
        Some(v) => v,
        None => _,
    };
}
