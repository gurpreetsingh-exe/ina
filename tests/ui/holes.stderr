error: type hole found
   ╭─[tests/ui/holes.ina:9:26]
   │
 9 │     let xs = [None, Some(_)];
   │                          ┬
   │                          ╰─ expected expression of type `i64`
   ∙
───╯
error: type hole found
    ╭─[tests/ui/holes.ina:12:17]
    │
 12 │     let _ = [x, _];
    │                 ┬
    │                 ╰─ expected expression of type `Option[i64]`
    ∙
────╯
error: type hole found
    ╭─[tests/ui/holes.ina:15:17]
    │
 15 │         None => _,
    │                 ┬
    │                 ╰─ expected expression of type `i8`
    ∙
────╯
